packages/image_picker/image_picker_for_web/lib/image_picker_for_web.dart

```diff
@@ -121,6 +121,9 @@ class ImagePickerPlugin extends ImagePickerPlatform {
     List<XFile> files = await getFiles(
       accept: _kAcceptImageMimeType,
       capture: capture,
+      maxWidth: maxWidth,
+      maxHeight: maxHeight,
+      imageQuality: imageQuality,
     );
     return files.first;
```

Comment:
Instead, determine if resizing is needed, and then `return imageResizer.resizeImage(files.first, maxWidth, maxHeight, imageQuality);`